clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -674,6 +662,7 @@ void transitionToSendingLeaveGroup() {
                 ConsumerGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH;
         updateMemberEpoch(leaveEpoch);
         currentAssignment = new HashMap<>();
+        targetAssignmentReconciled = currentAssignment.equals(currentTargetAssignment);
```

Comment:
I wonder if this could end up not being accurate in some case. Thinking about a consumer RECONCILING, and then leaving the group while on it. Here we would end up with `targetAssignmentReconciled` false, which is not accurate, as the target is not relevant anymore and will be discarded when that delayed reconciliation completes. 

Is there a value in the `targetAssignmentReconciled` var other than caching the sets comparison result? It definitely brings the challenge of making sure we update it with the right value, in all the right places. Would it be safer and simpler maybe just to keep the `targetAssignmentReconciled()` method, checking current vs target when needed?

Reply:
As for the first point - I don't think it will cause a problem, but you are right that the name of the boolean is not 100% precise in this case.

I introduced was mostly to save the overhead of having to compare two sets after every heartbeat. We can also not cache this for now, and see if we need to add it back later on. Premature optimization yadayada, I guess ;). I'll remove it for now.